cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -53,16 +57,67 @@
     this.artifactInspector = new ArtifactInspector(cConf, artifactClassLoaderFactory);
   }
 
-  @VisibleForTesting
-  void clear(Id.Namespace namespace) throws IOException {
+  /**
+   * Clear all artifacts in the given namespace. This method is only intended to be called by unit tests, and
+   * when a namespace is being deleted.
+   *
+   * @param namespace the namespace to delete artifacts in.
+   * @throws IOException if there was an error making changes in the meta store
+   */
+  public void clear(Id.Namespace namespace) throws IOException {
     artifactStore.clear(namespace);
   }
 
   /**
-   * Returns a {@link SortedMap} of all plugins available for the given artifact. The keys
+   * Get all artifacts in the given namespace, optionally including system artifacts as well. Will never return
+   * null. If no artifacts exist, an empty list is returned. Namespace existence is not checked.
+   *
+   * @param namespace the namespace to get artifacts from
+   * @param includeSystem whether system artifacts should be included in the results
+   * @return an unmodifiable list of artifacts that belong to the given namespace
+   * @throws IOException if there as an exception reading from the meta store
+   */
+  public List<ArtifactSummary> getArtifacts(Id.Namespace namespace, boolean includeSystem) throws IOException {
+    List<ArtifactSummary> summaries = Lists.newArrayList();
+    if (includeSystem) {
+      convertAndAdd(summaries, artifactStore.getArtifacts(Constants.SYSTEM_NAMESPACE_ID));
+    }
+    return Collections.unmodifiableList(convertAndAdd(summaries, artifactStore.getArtifacts(namespace)));
+  }
+
+  /**
+   * Get all artifacts in the given namespace of the given name. Will never return null.
+   * If no artifacts exist, an exception is thrown. Namespace existence is not checked.
```

Comment:
this one throws not found because the caller asks for a specific artifact by name, whereas the other just lists all artifacts.  corresponds to GET /artifacts vs GET /artifacts/etlbatch, with the first call never returning a 404 (because the 'artifacts' collection always exists) and the second possibly returning a 404.